plugins/module_utils/vm_device_helper.py

```diff
@@ -96,10 +103,10 @@ def create_nvme_controller(bus_number):
     def is_nvme_controller(device):
         return isinstance(device, vim.vm.device.VirtualNVMEController)
 
-    def create_disk_controller(self, ctl_type, ctl_number):
+    def create_disk_controller(self, ctl_type, ctl_number, bus_sharing='noSharing'):
         disk_ctl = None
-        if ctl_type in ['buslogic', 'paravirtual', 'lsilogic', 'lsilogicsas']:
-            disk_ctl = self.create_scsi_controller(ctl_type, ctl_number)
+        if ctl_type in self.scsi_device_typekeys():
```

Comment:
Is this "self.scsi_device_type.keys()"?

Reply:
Thanks @Tomorrow9! Yes, this looks like a typo and should be `self.scsi_device_type.keys()`. I've fixed it.

I wonder why this didn't fail the CI...

Reply:
@Tomorrow9!

> I wonder why this didn't fail the CI...

Now that's weird. I see `vmware_guest_controller` in the list of targets in the [log](https://c2e5685fdc38e21487ec-e80be9008b6e5d3fe56bd68b58f0d10f.ssl.cf2.rackcdn.com/926/ccb6366a605c800d6a7edb0699fd02a33fdce4c8/check/ansible-test-cloud-integration-vcenter7_1esxi-python36-stable210/3a911df/job-output.txt) but I don't see it being actually _run_.

Reply:
`create_disk_controller` function is not called in vmware_guest_controller?

Reply:
But if not add a new pvscsi controller, the below assert will fail, it's weird.
```
  - name: add new disk controllers to virtual machine
    vmware_guest_controller:
      validate_certs: false
      hostname: "{{ vcenter_hostname }}"
      username: "{{ vcenter_username }}"
      password: "{{ vcenter_password }}"
      name: "{{ virtual_machines[0].name }}"
      sleep_time: 30
      controllers:
        - type: sata
          state: present
        - type: paravirtual
          state: present
    register: add_disk_controller

  - debug: var=add_disk_controller

  - name: assert the new disk controllers were added to VM
    assert:
      that:
        - "add_disk_controller.changed == true"
        - "{{ add_disk_controller.disk_controller_data.scsi | length | int }} == {{ scsi_num | int + 1 }}"
        - "{{ add_disk_controller.disk_controller_data.sata | length | int }} == {{ sata_num | int + 1 }}"
 ```